packages/xstate-store/test/store.test.ts

```diff
@@ -339,6 +339,63 @@ it('emitted events occur after the snapshot is updated', () => {
   store.send({ type: 'inc' });
 });
 
+it('events can be emitted with no payload', () => {
+  const spy = jest.fn();
+
+  const store = createStore({
+    emits: {
+      incremented: () => {},
+      decremented: () => {},
+      expectsPayload: (_: { payload: string }) => {}
+    },
+    context: {
+      count: 0
+    },
+    on: {
+      inc: (_ctx, _ev, enq) => {
+        enq.emit.incremented();
+      },
+      dec: (_ctx, _ev, enq) => {
+        enq.emit.decremented(
+          // @ts-expect-error
+          {}
+        );
+      },
+      hasPayload: (_ctx, _ev, enq) => {
+        enq.emit
+          // @ts-expect-error
+          .expectsPayload();
+      }
+    }
+  });
+
+  store.on('incremented', spy);
+
+  store.send({ type: 'inc' });
+
+  expect(spy).toHaveBeenCalledWith({ type: 'incremented' });
+});
+
+it('events can be emitted with optional payloads (type check)', () => {
+  createStore({
+    emits: {
+      optionalPayload: (_: { payload?: string }) => {}
+    },
+    context: {},
+    on: {
+      inc: (_ctx, _ev, enq) => {
+        enq.emit
+          // @ts-expect-error
+          .optionalPayload();
```

Comment:
so that was kind of my question - should this be an error if all fields are optional?

Reply:
I don't think so; what would be the correct type in this case?

Reply:
but is requiring an empty object to be passed in natural? fwiw you can achieve conditional parameter optionality like this: [TS playground](https://www.typescriptlang.org/play/?ts=5.9.0-dev.20250317#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwFsoBrEAHgBV4QAPDEVYAZ3gG94MBPABxAC54TDDCyoA5vAC+APgAUASgGyAUPHgA6TbDFMB7LrwEUA2gCIDIUwF0p1Og2bwKqtfAD88UpxyInd+ows5jyWLq7u8MZWYa4CxtxQnBA4UMBuAgDyBFgYlAA08MG8ptLR4XEJSSnAmdm5FAVFlqXK8vAAvNLwAG44WMAA3MrKYHhCgg4AjB2EJOT6IQKmAGIZGaZScvJDykxTCkN7jJOyrJLbu-sLhoWr61IXI2MYE4wATDNEpGTX-LdrpgG8AARrABKhkARgSAYJsDsMjsA3vDEcizhdUacQWD4NNzjtRqhxoiAMyfOY-DiLf7rIGgmDpeAQqEwuGPUkohwk074y6MbnsekCPEXIA)